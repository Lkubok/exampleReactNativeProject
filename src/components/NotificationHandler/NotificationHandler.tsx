/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable @typescript-eslint/unbound-method */
import React, { FC, useEffect, useCallback } from 'react';
import { View } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { useDispatch } from 'react-redux';

import { usePushNotification } from 'hooks';
import { screenNames } from 'navigation';
import { actions as offerActions } from 'store/offers';

const NotificationHandler: FC = () => {
  const {
    notificationType,
    clearNotificationData,
    notificationData,
    setTriggerMarketPlaceRefresh,
  } = usePushNotification();
  const { navigate } = useNavigation();
  const dispatch = useDispatch();

  console.log('TEST', notificationData, notificationType);

  const handleNewMessageArrive = useCallback(() => {
    dispatch(offerActions.setCurrentOfferData(null));
    console.log('NOT DATA', notificationData);
    navigate(screenNames.MainStackNames.NewsStack, {
      screen: screenNames.MainFlow.NewsStack.NewsDetail,
      params: { customer_slug: notificationData?.customer_slug, offer_slug: notificationData?.offer_slug },
    });
    clearNotificationData();
  }, [clearNotificationData, dispatch, navigate, notificationData]);

  const handleUserCanEvaluate = useCallback(() => {
    navigate(screenNames.MainStackNames.NewsStack, {
      screen: screenNames.MainFlow.NewsStack.NewsDetail,
      params: { customer_slug: notificationData?.customer_slug, offer_slug: notificationData?.offer_slug },
    });
    clearNotificationData();
  }, [clearNotificationData, navigate, notificationData?.customer_slug, notificationData?.offer_slug]);

  const handleUserWasEvaluated = useCallback(() => {
    navigate(screenNames.MainStackNames.NewsStack, {
      screen: screenNames.MainFlow.NewsStack.NewsDetail,
      params: { customer_slug: notificationData?.customer_slug, offer_slug: notificationData?.offer_slug },
    });
    clearNotificationData();
  }, [clearNotificationData, navigate, notificationData?.customer_slug, notificationData?.offer_slug]);

  const handleNewOffersInMarketPlace = useCallback(() => {
    setTriggerMarketPlaceRefresh(true);
    navigate(screenNames.MainStackNames.MarketPlaceStack, {
      screen: screenNames.MainFlow.MarketPlaceStack.MarketPlaceDashboard,
    });
    clearNotificationData();
  }, [clearNotificationData, navigate, setTriggerMarketPlaceRefresh]);

  const launchPushNotificationHandler = useCallback(() => {
    if (notificationType) {
      switch (notificationType) {
        case 'new_message':
          handleNewMessageArrive();
          break;
        case 'scoring_user_can_evaluate':
          handleUserCanEvaluate();
          break;
        case 'scoring_user_was_evaluated':
          handleUserWasEvaluated();
          break;
        case 'new_offers_in_marketplace':
          handleNewOffersInMarketPlace();
          break;
        default:
          break;
      }
    }
  }, [
    handleNewMessageArrive,
    handleNewOffersInMarketPlace,
    handleUserCanEvaluate,
    handleUserWasEvaluated,
    notificationType,
  ]);

  useEffect(() => {
    if (notificationType) {
      launchPushNotificationHandler();
    }
  }, [launchPushNotificationHandler, notificationType]);

  return <View />;
};

export default NotificationHandler;
