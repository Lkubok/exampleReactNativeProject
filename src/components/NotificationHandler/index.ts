import Component from './NotificationHandler';

export default Component;
