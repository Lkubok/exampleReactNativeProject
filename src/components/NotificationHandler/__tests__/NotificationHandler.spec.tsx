import React from 'react';
import { act, customRender } from 'utils/testUtils';

import NotificationHandler from '../NotificationHandler';

jest.useFakeTimers();

describe('NotificationHandler component', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let props: any;
  beforeEach(() => {
    props = {
      disabled: true,
      loading: false,
      onPress: () => {},
      title: 'testingNotificationHandler',
      useNativeDriver: true,
    };
  });

  it('should match snapshot', () => {
    const snapshot = customRender(<NotificationHandler {...props} />).toJSON();
    // eslint-disable-next-line no-void
    void act(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
