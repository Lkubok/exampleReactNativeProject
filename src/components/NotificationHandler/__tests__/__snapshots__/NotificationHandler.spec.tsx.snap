// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationHandler component should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View />
</RNCSafeAreaProvider>
`;
